<html><h1>Validation FAILED.</h1><p>Build Time: 20 minutes 39 seconds 797 ms</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/com-liferay-commerce/tree/7.1.x">7.1.x</a><br/>Branch GIT ID: <a href="https://github.com/liferay/com-liferay-commerce/commit/33857084450ec106ccb267826ce2acaf16e46be2">33857084450ec106ccb267826ce2acaf16e46be2</a></p><h6>Task Summary:</h6><ul><li>subrepository-compile-jdk8 -  :heavy_check_mark:</li><li>subrepository-semantic-versioning-jdk8 -  :heavy_check_mark:</li><li>subrepository-pmd-jdk8 -  :heavy_check_mark:</li><li>subrepository-source-format-jdk8 -  :x:<pre><code>     [echo] java.lang.Exception: Found 4 formatting issues:
     [echo] 1: ./commerce-product-content-category-web/build.gradle expected:&lt;...om.liferay.commerce.[media.api", version: "1.1.0"
     [echo] 	compileOnly group: "com.liferay.commerce", name: "com.liferay.commerce.product.api", version: "6.0].0"
     [echo] 	compileOnly gro...&gt; but was:&lt;...om.liferay.commerce.[product.api", version: "6.0.0"
     [echo] 	compileOnly group: "com.liferay.commerce", name: "com.liferay.commerce.media.api", version: "1.1].0"
     [echo] 	compileOnly gro...&gt;
     [echo] 2: ./commerce-product-asset-categories-web/build.gradle expected:&lt;...om.liferay.commerce.[media.api", version: "1.1.0"
     [echo] 	compileOnly group: "com.liferay.commerce", name: "com.liferay.commerce.product.api", version: "6.0].0"
     [echo] 	compileOnly gro...&gt; but was:&lt;...om.liferay.commerce.[product.api", version: "6.0.0"
     [echo] 	compileOnly group: "com.liferay.commerce", name: "com.liferay.commerce.media.api", version: "1.1].0"
     [echo] 	compileOnly gro...&gt;
     [echo] 3: ./commerce-product-asset-categories-navigation-web/build.gradle expected:&lt;...om.liferay.commerce.[media.api", version: "1.1.0"
     [echo] 	compileOnly group: "com.liferay.commerce", name: "com.liferay.commerce.product.api", version: "6.0].0"
     [echo] 	compileOnly gro...&gt; but was:&lt;...om.liferay.commerce.[product.api", version: "6.0.0"
     [echo] 	compileOnly group: "com.liferay.commerce", name: "com.liferay.commerce.media.api", version: "1.1].0"
     [echo] 	compileOnly gro...&gt;
     [echo] 4: ./commerce-product-definitions-web/build.gradle expected:&lt;...om.liferay.commerce.[media.api", version: "1.1.0"
     [echo] 	compileOnly group: "com.liferay.commerce", name: "com.liferay.commerce.product.api", version: "6.0].0"
     [echo] 	compileOnly gro...&gt; but was:&lt;...om.liferay.commerce.[product.api", version: "6.0.0"
     [echo] 	compileOnly group: "com.liferay.commerce", name: "com.liferay.commerce.media.api", version: "1.1].0"
     [echo] 	compileOnly gro...&gt;
     [echo] 
     [echo] 	at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:430)
     [echo] 	at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:258)
     [echo] 
     [echo] FAILURE: Build failed with an exception.
     [echo] 
     [echo] * What went wrong:
     [echo] Execution failed for task ':checkSourceFormatting'.</code></pre></li><li>central-requirements-jdk8/subrepository-unit-jdk8 -  :heavy_check_mark:</li></ul><h5>For full console, click <a href="https://test-1-18.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/28//consoleText">here</a>.</h5><hr/><div><h6>Test Results:</h6><div><div><p>46 Tests Passed.<br/>0 Tests Failed.</p></div></div><h5>For all test results, click <a href="https://test-1-18.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/28//testReport">here</a>.</h5></div></html>

--- a/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPDefinitionLinkMVCActionCommand.java
+++ b/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPDefinitionLinkMVCActionCommand.java
@@ -15,28 +15,21 @@
 package com.liferay.commerce.product.definitions.web.internal.portlet.action;
 
 import com.liferay.commerce.product.constants.CPPortletKeys;
-import com.liferay.commerce.product.definitions.web.servlet.taglib.ui.CPDefinitionScreenNavigationConstants;
 import com.liferay.commerce.product.exception.NoSuchCPDefinitionLinkException;
-import com.liferay.commerce.product.model.CPDefinition;
 import com.liferay.commerce.product.model.CPDefinitionLink;
 import com.liferay.commerce.product.service.CPDefinitionLinkService;
-import com.liferay.portal.kernel.portlet.PortletProvider;
-import com.liferay.portal.kernel.portlet.PortletProviderUtil;
 import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
 import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
 import com.liferay.portal.kernel.security.auth.PrincipalException;
 import com.liferay.portal.kernel.service.ServiceContext;
 import com.liferay.portal.kernel.service.ServiceContextFactory;
 import com.liferay.portal.kernel.servlet.SessionErrors;
-import com.liferay.portal.kernel.theme.ThemeDisplay;
 import com.liferay.portal.kernel.util.Constants;
 import com.liferay.portal.kernel.util.ParamUtil;
 import com.liferay.portal.kernel.util.StringUtil;
-import com.liferay.portal.kernel.util.WebKeys;
 
 import javax.portlet.ActionRequest;
 import javax.portlet.ActionResponse;
-import javax.portlet.PortletURL;
 
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
@@ -122,7 +115,7 @@ public class EditCPDefinitionLinkMVCActionCommand extends BaseMVCActionCommand {
 				updateCPDefinitionLink(actionRequest);
 			}
 
-			String redirect = getSaveAndContinueRedirect(actionRequest);
+			String redirect = ParamUtil.getString(actionRequest, "redirect");
 
 			sendRedirect(actionRequest, actionResponse, redirect);
 		}
@@ -138,47 +131,6 @@ public class EditCPDefinitionLinkMVCActionCommand extends BaseMVCActionCommand {
 				throw e;
 			}
 		}
-	}
-
-	protected String getSaveAndContinueRedirect(ActionRequest actionRequest)
-		throws Exception {
-
-		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
-			WebKeys.THEME_DISPLAY);
-
-		PortletURL portletURL = PortletProviderUtil.getPortletURL(
-			actionRequest, themeDisplay.getScopeGroup(),
-			CPDefinition.class.getName(), PortletProvider.Action.EDIT);
-
-		long cpDefinitionLinkId = ParamUtil.getLong(
-			actionRequest, "cpDefinitionLinkId");
-
-		if (cpDefinitionLinkId > 0) {
-			portletURL.setParameter(
-				"cpDefinitionLinkId", String.valueOf(cpDefinitionLinkId));
-		}
-
-		portletURL.setParameter(
-			"mvcRenderCommandName", "editProductDefinition");
-
-		long cpDefinitionId = ParamUtil.getLong(
-			actionRequest, "cpDefinitionId");
-
-		portletURL.setParameter(
-			"cpDefinitionId", String.valueOf(cpDefinitionId));
-
-		String toolbarItem = ParamUtil.getString(actionRequest, "toolbarItem");
-
-		portletURL.setParameter("toolbarItem", toolbarItem);
-
-		portletURL.setParameter(
-			"screenNavigationCategoryKey",
-			CPDefinitionScreenNavigationConstants.
-				CATEGORY_KEY_PRODUCT_RELATIONS);
-
-		portletURL.setWindowState(actionRequest.getWindowState());
-
-		return portletURL.toString();
 	}
 
 	protected CPDefinitionLink updateCPDefinitionLink(

